src/lib/coda_base/transition_system.ml

```diff
@@ -149,6 +152,8 @@ struct
     let%snarkydef main (logger : Logger.t) (top_hash : Digest.Tick.Packed.var)
         =
       let%bind prev_state = exists' State.typ ~f:Prover_state.prev_state
+      and genesis_state_hash =
```

Comment:
I'm not totally sure this makes sense -- as is I think it's just saying 

```ocaml
let is_base_case = 
  choose genesis_state_hash in genesis_state_hash = next_state_hash 
```
so the prover can pick `genesis_state_hash = next_state_hash` to make `is_base_case = true` or can pick it equal to some other value to make `is_base_case = false`, no?

Reply:
Oh you're right. So `next_state_hash` should be the `genesis_state_hash` then?

Reply:
Umm, actually, the prover can pick whatever value for genesis_state but it'll verify only with the same genesis state and so that way the blocks are guaranteed to be from the same genesis state. But I suppose this can be achieved by making the `next_state` equal to genesis_state?

Reply:
Ok -- I think I realized what I was confused about. I think that the concept of "genesis_state_hash" does not belong in this module.

That is, it should be something internal to the blockchain_snark_state state machine and this module should just ask that module if it's a base case, which it can do with this logic. Does that make sense?